modules/public/public.go

```diff
@@ -19,15 +25,134 @@ import (
 // Options represents the available options to configure the macaron handler.
 type Options struct {
 	Directory   string
+	IndexFile   string
 	SkipLogging bool
+	// if set to true, will enable caching. Expires header will also be set to
+	// expire after the defined time.
+	ExpiresAfter time.Duration
+	FileSystem   http.FileSystem
+	Prefix       string
 }
 
 // Custom implements the macaron static handler for serving custom assets.
 func Custom(opts *Options) macaron.Handler {
-	return macaron.Static(
-		path.Join(setting.CustomPath, "public"),
-		macaron.StaticOptions{
-			SkipLogging: opts.SkipLogging,
-		},
-	)
+	return opts.staticHandler(path.Join(setting.CustomPath, "public"))
+}
+
+// staticFileSystem implements http.FileSystem interface.
+type staticFileSystem struct {
+	dir *http.Dir
+}
+
+func newStaticFileSystem(directory string) staticFileSystem {
+	if !filepath.IsAbs(directory) {
+		directory = filepath.Join(macaron.Root, directory)
+	}
+	dir := http.Dir(directory)
+	return staticFileSystem{&dir}
+}
+
+func (fs staticFileSystem) Open(name string) (http.File, error) {
+	return fs.dir.Open(name)
+}
+
+// StaticHandler sets up a new middleware for serving static files in the
+func StaticHandler(dir string, opts *Options) macaron.Handler {
+	return opts.staticHandler(dir)
+}
+
+func (opts *Options) staticHandler(dir string) macaron.Handler {
```

Comment:
Most of staticHandler has been copied over from parts of macaron. The handle function was modified to use ExpireAfter instead of the Expires function, and also now it has proper support for ETags (by reading the req header `If-None-Match`).